cfn-resources/markdown-generator/generate_version_markdown.go

```diff
@@ -0,0 +1,136 @@
+		"ap-south-2", "ap-southeast-4", "eu-central-2", "eu-south-2", "me-central-1"}
+
+	var resourceTypes = GetTypeNamesFromRpdkJSONFiles()
+
+	// map of resource type name to list of values where each index corresponds to a region
+	resourceVersions := obtainVersions(regions, resourceTypes)
+	_ = generateMarkdown(ouputMarkdownFile, regions, resourceTypes, resourceVersions)
+}
+
+func obtainVersions(regions, resourceTypes []string) map[string][]string {
+	resourceValues := map[string][]string{}
+
+	for _, typeName := range resourceTypes {
+		resourceValues[typeName] = make([]string, len(regions))
+	}
+
+	for i, region := range regions {
+		typeSummaries := getTypeSummariesOfRegion(region)
+
+		for _, typeSummary := range typeSummaries {
+			fmt.Printf("Region: %s, type name: %s, version: %s, date: %s\n", region, *typeSummary.TypeName, *typeSummary.LatestPublicVersion, typeSummary.LastUpdated.String())
+			if _, exists := resourceValues[*typeSummary.TypeName]; exists {
+				if typeSummary.LatestPublicVersion != nil && typeSummary.LastUpdated != nil {
+					date := *typeSummary.LastUpdated
+					resourceValues[*typeSummary.TypeName][i] = fmt.Sprintf("%s (%s)", *typeSummary.LatestPublicVersion, date.Format("2006-01-02"))
+				}
+			}
+		}
+	}
+	return resourceValues
+}
+
+func generateMarkdown(filename string, regions, resourceTypes []string, resourceValues map[string][]string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	for i, r := range regions {
+		regions[i] = useNoBreakChar(r)
+	}
+
+	headers := []string{"Latest version"}
+	headers = append(headers, regions...)
+	_, _ = fmt.Fprintf(file, "%s\n", strings.Join(headers, "|"))
+
+	divider := make([]string, len(headers))
+	for i := range divider {
+		divider[i] = "-"
+	}
+	_, _ = fmt.Fprintf(file, "%s\n", strings.Join(divider, "|"))
+
+	for _, typeName := range resourceTypes {
+		row := []string{fmt.Sprintf("**%s**", typeName)}
+		for regionIndex := range regions {
+			value := resourceValues[typeName][regionIndex]
+			row = append(row, useNoBreakChar(value))
+		}
+		_, _ = fmt.Fprintf(file, "%s\n", strings.Join(row, "|"))
+	}
+	return nil
+}
+
+func getTypeSummariesOfRegion(region string) []types.TypeSummary {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	cfg.Region = region
+
+	if err != nil {
+		log.Fatalf("failed to load SDK configuration, %v", err)
+	}
+
+	client := cloudformation.NewFromConfig(cfg)
+
+	input := &cloudformation.ListTypesInput{
+		Visibility: types.VisibilityPublic,
+		Type:       types.RegistryTypeResource,
+		Filters: &types.TypeFilters{
+			PublisherId: aws.String(mongoDBPublisherID),
+		},
+	}
+
+	paginator := cloudformation.NewListTypesPaginator(client, input)
+
+	results := []types.TypeSummary{}
+	// Iterate through paginated results
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(context.Background())
+		if err != nil {
```

Comment:
```suggestion
		if page, err := paginator.NextPage(context.Background()); err != nil {
```

Reply:
nit

Reply:
checked this change but does not bring much benefit as `page` is used outside of if statement and therefore needs to be defined before.